frontend/src/plugins/i18n/apiFallbackLocalesFor.js

```diff
@@ -1,13 +1,34 @@
 import { fallbackLocale } from '@/plugins/i18n/index'
 
-export default function* (locale) {
-  if (typeof locale !== 'string') {
-    yield fallbackLocale
-    return
+export default function* (locale, separator = '_') {
+  if (typeof locale === 'string') {
+    const parts = locale.split(separator)
+    for (let i = parts.length - 1; i > 0; i--) {
+      let slice = parts.slice(0, i)
+      yield slice.join(separator)
+      if (slice.join('-') in fallbackLocale) {
+        for (const fallback of fallbackLocale[slice.join('-')]) {
+          yield fallback.split('-').join(separator)
+        }
+      }
+    }
+    if (locale.split(separator).join('-') in fallbackLocale) {
+      for (const fallback of fallbackLocale[locale.split(separator).join('-')]) {
+        yield fallback.split('-').join(separator)
+      }
+    }
   }
-  const parts = locale.split('_')
-  for (let i = parts.length - 1; i > 0; i--) {
-    yield parts.slice(0, i).join('_')
+  if (typeof fallbackLocale === 'string') {
+    yield fallbackLocale
+  } else if (Array.isArray(fallbackLocale)) {
+    for (const fallback of fallbackLocale) {
+      yield fallback
+    }
+  } else {
```

Comment:
Why do we want to start supporting string and Array fallback locales? You are importing `fallbackLocale` from a single fixed file in our repo. Is this a case of YAGNI?

Reply:
done

Reply:
But isn't this the usecase for rumantsch where you want german as a fallback of `rm-CH-scout`? Otherwise it will just be `rm-CH` and `rm`